lib/live_cache.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+# This is a replacement for DistributedCache which is bounded and respects
+# clears during deploys
+class LiveCache
+  class Manager
+    CHANNEL_NAME ||= "/live_cache"
+
+    def initialize(message_bus: MessageBus)
+      @message_bus = message_bus
+      @subscription = Concurrent::Promises.delay(&method(:subscribe))
+      @mutex = Mutex.new
+      @caches = Hash.new { |h, k| h[k] = WeakList.new }
+    end
+
+    def register(hash_key, cache)
+      @mutex.synchronize { @caches[hash_key] << cache }
+
+      ensure_subscribed!
+    end
+
+    def publish(message)
+      @message_bus.publish(CHANNEL_NAME, message)
+    end
+
+    private
+
+    def process_message(message)
+      @mutex.synchronize do
+        @caches[message.data["hash_key"]].each { |cache| cache.process_message(message) }
+      end
+    end
+
+    def subscribe
+      @message_bus.subscribe(CHANNEL_NAME) { |message| process_message(message) }
+    end
+
+    def ensure_subscribed!
+      @subscription.wait!
+    end
+  end
+
+  class << self
+    attr_accessor :default_manager
+  end
+
+  self.default_manager = Manager.new
+
+  attr_reader :identity
+
+  def initialize(
+    hash_key,
+    max_size_per_site,
```

Comment:
```suggestion
    max_size_per_site:,
```

I think using a keyword arg here can help improve readability when the cache is being initialised